drivers/interrupt_controller/intc_plic.c

```diff
@@ -50,17 +102,42 @@ static int save_irq;
  *
  * @return irq value when enabled 0 otherwise
  */
-static int riscv_plic_is_edge_irq(uint32_t irq)
+static int riscv_plic_is_edge_irq(uint32_t irq, const struct device *dev)
 {
-	if (IS_ENABLED(CONFIG_PLIC_SUPPORTS_EDGE_IRQ)) {
-		volatile uint32_t *trig = (volatile uint32_t *)PLIC_EDGE_TRIG_TYPE;
+	const struct plic_config *config = get_plic_config(dev);
+
+	if (config->trig != 0) {
+		volatile uint32_t *trig = (volatile uint32_t *) config->trig;
 
 		trig += (irq >> PLIC_EDGE_TRIG_SHIFT);
 		return *trig & BIT(irq);
 	}
 	return 0;
 }
 
+void z_riscv_plic_isr_intc_install(const struct device *intc, unsigned int irq, unsigned int priority,
+			      void (*routine)(const void *), const void *param)
+{
+	__ASSERT(!irq_is_enabled(irq), "IRQ %d is enabled", irq);
+
+	unsigned int table_idx = z_get_sw_isr_table_idx(irq) + get_isr_table_offset(intc);
+
+	_sw_isr_table[table_idx].arg = param;
+	_sw_isr_table[table_idx].isr = routine;
+
+	if (irq_get_level(irq) == 2) {
+		irq = irq_from_level_2(irq);
+
+		riscv_plic_intc_set_priority(intc, irq, priority);
+	}
+}
+
+void z_riscv_plic_isr_install(unsigned int irq, unsigned int priority,
+			      void (*routine)(const void *), const void *param)
+{
+	z_riscv_plic_isr_intc_install(NULL, irq, priority, routine, param);
```

Comment:
Here, we have the ability to get `dev` upfront, so might as well do that, and then it doesn't need to be done in `z_riscv_plic_isr_intc_install()`